### DISCUSSION TITLE
Add a linter / code check

### DISCUSSION THREAD

**Discussion by thebonbon** [EveronLife/EveronLife#64] [Ideas]
A linter / code check should be added as soon as possible. 
It should be checking every PR for programming errors / styles.

Wiki:
> Lint, or a linter, is a [static code analysis](https://en.wikipedia.org/wiki/Static_program_analysis) tool used to flag programming errors, [bugs](https://en.wikipedia.org/wiki/Software_bug), stylistic errors and suspicious constructs.

Example:
https://github.com/github/super-linter

**Comment by Arkensor** [comment]
Yep, we will want to look into this. We will give it a few weeks to gather experience so we can lock down conventions for all file types and then we would want to automate the linting process for all PRs.

**Reply by thebonbon** [reply]
Better to implement it sooner than later. Would need to change all code in a few weeks again. Why not do it right from the beginning?
Scripting conventions are set in the official reforger wiki. We already reference them in the style guide.
https://github.com/EveronLife/EveronLife/blob/main/.github/CONTRIBUTING.md#-code-styleguide

What experience do you want to gather? Nothing is really gonna change

**Reply by Arkensor** [reply]
We do not have a linter to support enscript and it will be a while before one is made that satisfies all our needs. So until then we can not automatically enforce it. That is what I was wanting to look into.

The experiences are for situations that are not covered by the existing rules. Because for an auto check on PR's we will want to check filenames (with potential prefixes) and their locations. And for this we lack a significant number of use cases to already decide on what structure and naming will be suitable. There are open problems such as #73
